### PR TITLE
chore: update the go version and dependencies

### PR DESCRIPTION
**Pull request by ishitasequeira**
**What type of PR is this?**
 /kind chore

**What does this PR do / why we need it**:
Upgrade go version to 1.18 and go.mod dependencies

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:
Execute below commands in different terminal windows:
* make install run
* make test

### CODE REVIEW

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (ishitasequeira) [inline api/v1alpha1/zz_generated.deepcopy.go:1]: `make generate` automatically seems to be adding this now. If we remove this line, the ci fails. Not sure if there is a workaround for this.

Comment (jopit) [reply api/v1alpha1/zz_generated.deepcopy.go:1]: It's actually the `go fmt` command that's doing this as of golang 1.17: https://stackoverflow.com/questions/10646531/golang-conditional-compilation/67937234#67937234

If we're moving to golang 1.18, we shouldn't need to work around this.

**Review by sagikazarmark — APPROVED** [review]
Tested the changes to the toolchain locally. Works for me on Go 1.18

**Comment by iam-veeramalla** [comment]
@ishitasequeira there are many manifests file changes introduced by this PR. Are all of them expected ? Can you please re-verify  ?

**Comment by ishitasequeira** [comment]
> @ishitasequeira there are many manifests file changes introduced by this PR. Are all of them expected ? Can you please re-verify ?

@iam-veeramalla, the manifests are auto-updated after executing `make bundle`.

**Comment by jannfis** [comment]
> the manifests are auto-updated after executing

I think this is due to the update of K8s dependencies.

File: docs/reference/argocd.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ Name | Default | Description
 [**StatusBadgeEnabled**](#status-badge-enabled) | `true` | Enable application status badge feature.
 [**TLS**](#tls-options) | [Object] | TLS configuration options.
 [**UsersAnonymousEnabled**](#users-anonymous-enabled) | `true` | Enable anonymous user access.
-[**Version**](#version) | v2.2.2 (SHA) | The tag to use with the container image for all Argo CD components.
```
Comment (jannfis) [inline docs/reference/argocd.md:47]: The version bump of the image is not part of this change. Has this been done before, and this is just docs update?

Comment (ishitasequeira) [reply docs/reference/argocd.md:47]: Yes, We changed the container image while releasing 0.3.0.

File: api/v1alpha1/zz_generated.deepcopy.go (hunk 1 above)
Comment (jaideepr97) [reply api/v1alpha1/zz_generated.deepcopy.go:1]: I find that this happens when I use vscode
whenever I need to get rid of this I edit the file in the native fedora editor and that sticks

**Comment by jaideepr97** [comment]
LGTM, thanks @ishitasequeira !

**Comment by rishabh625** [comment]
It LGTM except that docs could be changed in sperate PR , @jannfis : can u plz take a look ?

**Review by jannfis — APPROVED** [review]
LGTM
